### PR TITLE
Use QDoubleRangeSlider from superqt package

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
[EDIT] This PR swaps out our QRangeSlider for the QDoubleRangeSlider in [superqt](https://github.com/napari/superqt) 

## Type of change

- [ ] Bug-fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

**Comment by sofroniewn** [comment]
Should we review/ merge this before #2753? Does that PR subsume this one?

Looks like it's got a couple test fails, could be upstream. Happy to give this a review when ready.

**Comment by sofroniewn** [comment]
Ah i just realized that #2753 is about labels on the other sliders, not about range - i see

File: napari/_qt/layer_controls/qt_image_controls_base.py
Comment (ppwadhwa) [inline napari/_qt/layer_controls/qt_image_controls_base.py:80]: Can we add a tooltip here to let the user know that there is a pop up?  I think it would be helpful for those that don't know about the pop up.
Something similar to--
```self.contrastLimitsSlider.setToolTip(trans._('Right click for pop up contrast limits slider.'))```

**Review by ppwadhwa — APPROVED** [review]
Looks good to me!  I just had one request about the tooltip, but other than that it looks and works great!

File: napari/_qt/layer_controls/qt_image_controls_base.py
Comment (sofroniewn) [reply napari/_qt/layer_controls/qt_image_controls_base.py:80]: I like that idea!!

**Comment by sofroniewn** [comment]
Aside from failing tests this one looks good to me, will be happy to approve and merge when ready

**Comment by tlambert03** [comment]
both this one and #2753 are waiting on some upstream fixes to QDoubleRangeSlider ... hold off for now

File: napari/_qt/layer_controls/qt_image_controls_base.py
Comment (tlambert03) [reply napari/_qt/layer_controls/qt_image_controls_base.py:80]: added!

**Review by sofroniewn — APPROVED** [review]
Works great for me, just tested and all functionality works and looks same as before! Very nice

File: napari/_qt/widgets/qt_range_slider_popup.py
Comment (jni) [inline napari/_qt/widgets/qt_range_slider_popup.py]:
```suggestion
        """A popup window that contains a labeled range slider and buttons.
```

**Review by jni — APPROVED** [review]
Awesome! Love these changes.

**Pull request merged**
